transport.go

```diff
@@ -4,70 +4,51 @@ import (
 	"bufio"
 	"context"
 	"errors"
-	"log"
+	"fmt"
 	"net/http"
 	"strconv"
 	"strings"
-	"time"
 
 	"github.com/SkycoinProject/dmsg"
 	"github.com/SkycoinProject/dmsg/cipher"
-	"github.com/SkycoinProject/dmsg/disc"
 )
 
-// Defaults for dmsg configuration, such as discovery URL
-const (
-	DefaultDiscoveryURL = "http://dmsg.discovery.skywire.cc"
-)
-
-// DMSGTransport holds information about client who is initiating communication.
-type DMSGTransport struct {
-	Discovery  disc.APIClient
-	PubKey     cipher.PubKey
-	SecKey     cipher.SecKey
-	RetryCount uint8
+// Transport holds information about client who is initiating communication.
+type Transport struct {
+	DmsgClient *dmsg.Client
 }
 
 // RoundTrip implements golang's http package support for alternative transport protocols.
-// In this case DMSG is used instead of TCP to initiate the communication with the server.
-func (t DMSGTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+// In this case dmsg is used instead of TCP to initiate the communication with the server.
+func (t Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// process remote pub key and port from dmsg-addr request header
 	addrSplit := strings.Split(req.Host, ":")
 	if len(addrSplit) != 2 {
-		return nil, errors.New("Invalid server Pub Key or Port")
+		return nil, errors.New("invalid server Pub Key or Port")
 	}
 	var pk cipher.PubKey
 	if err := pk.Set(addrSplit[0]); err != nil {
 		return nil, err
 	}
-	rPort, _ := strconv.Atoi(addrSplit[1])
-	port := uint16(rPort)
 
-	serverAddress := dmsg.Addr{PK: pk, Port: port}
-	dmsgC, err := getClient(t.PubKey, t.SecKey)
+	rPort, err := strconv.Atoi(addrSplit[1])
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid port: %v", err)
 	}
+	port := uint16(rPort)
+
+	serverAddress := dmsg.Addr{PK: pk, Port: port}
 
 	var (
 		stream    *dmsg.Stream
 		streamErr error
 	)
-	for i := uint8(0); i < t.RetryCount; i++ {
-		stream, streamErr = dmsgC.DialStream(context.Background(), serverAddress)
-		if streamErr != nil {
-			log.Printf("Error dialing responder: %s. retrying...", streamErr)
-			// Adding this to make sure we have enough time for delegate servers to become available
-			time.Sleep(200 * time.Millisecond)
-			continue
-		}
-		streamErr = nil
-		break
-	}
+
+	stream, streamErr = t.DmsgClient.DialStream(context.Background(), serverAddress)
```

Comment:
Regarding this retry count, I'm sure this Dial was breaking back with a dmsg milestone1 implementation when we have multiple requests coming in on CXO (that is using dmsghttp to communicate). 
Sometimes it successfully dialed only after more then 10 retries when server is busy. Occasionally  dmsg server was not even under heavy load when this dial was breaking, but again that was with different implementation present in milestone1.

Maybe it would be reasonable to add a test case with a couple of parallel request to make sure each of them is served without this RetryCount gimmick.